### ISSUE TITLE
[Component] [focus-trap] Not able to select value from date picker which is triggered from a dialog

### ISSUE THREAD

**Issue by hellodresji** [element-plus/element-plus#10236]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.41`
- Element Plus Version: `2.2.19`
- Browser / OS: `Chrome/106.0.0.0`
- Build Tool: `Webpack`

### Related Component

- `el-focus-trap`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uIHRleHQgQGNsaWNrPVwiZGlhbG9nVmlzaWJsZSA9IHRydWVcIj5cbiAgICBjbGljayB0byBvcGVuIHRoZSBEaWFsb2dcbiAgPC9lbC1idXR0b24+XG5cbiAgPGVsLWRpYWxvZ1xuICAgIHYtbW9kZWw9XCJkaWFsb2dWaXNpYmxlXCJcbiAgICB0aXRsZT1cIlRpcHNcIlxuICAgIHdpZHRoPVwiMzAlXCJcbiAgICA6YmVmb3JlLWNsb3NlPVwiaGFuZGxlQ2xvc2VcIlxuICA+XG4gICAgPHNwYW4+VGhpcyBpcyBhIG1lc3NhZ2U8L3NwYW4+XG4gICAgPGRpdj5cbiAgICBcdDxlbC1kYXRlLXBpY2tlcj48L2VsLWRhdGUtcGlja2VyPlxuICAgIDwvZGl2PlxuICAgIDx0ZW1wbGF0ZSAjZm9vdGVyPlxuICAgICAgPHNwYW4gY2xhc3M9XCJkaWFsb2ctZm9vdGVyXCI+XG4gICAgICAgIDxlbC1idXR0b24gQGNsaWNrPVwiZGlhbG9nVmlzaWJsZSA9IGZhbHNlXCI+Q2FuY2VsPC9lbC1idXR0b24+XG4gICAgICAgIDxlbC1idXR0b24gdHlwZT1cInByaW1hcnlcIiBAY2xpY2s9XCJkaWFsb2dWaXNpYmxlID0gZmFsc2VcIj5cbiAgICAgICAgICBDb25maXJtXG4gICAgICAgIDwvZWwtYnV0dG9uPlxuICAgICAgPC9zcGFuPlxuICAgIDwvdGVtcGxhdGU+XG4gIDwvZWwtZGlhbG9nPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IEVsTWVzc2FnZUJveCB9IGZyb20gJ2VsZW1lbnQtcGx1cydcblxuY29uc3QgZGlhbG9nVmlzaWJsZSA9IHJlZihmYWxzZSlcblxuY29uc3QgaGFuZGxlQ2xvc2UgPSAoZG9uZTogKCkgPT4gdm9pZCkgPT4ge1xuICBFbE1lc3NhZ2VCb3guY29uZmlybSgnQXJlIHlvdSBzdXJlIHRvIGNsb3NlIHRoaXMgZGlhbG9nPycpXG4gICAgLnRoZW4oKCkgPT4ge1xuICAgICAgZG9uZSgpXG4gICAgfSlcbiAgICAuY2F0Y2goKCkgPT4ge1xuICAgICAgLy8gY2F0Y2ggZXJyb3JcbiAgICB9KVxufVxuPC9zY3JpcHQ+XG48c3R5bGUgc2NvcGVkPlxuLmRpYWxvZy1mb290ZXIgYnV0dG9uOmZpcnN0LWNoaWxkIHtcbiAgbWFyZ2luLXJpZ2h0OiAxMHB4O1xufVxuPC9zdHlsZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

### Steps to reproduce

1. render date picker in a dialog
2. open dialog and trigger the date picker
![image](https://user-images.githubusercontent.com/13019006/197803146-dc7e591a-3ee8-4aec-a465-5845cbded9be.png)
4. try to select a date from date picker panel
![image](https://user-images.githubusercontent.com/13019006/197803224-a1058157-c1d1-449e-87e5-c4fe04d0047a.png)
Nothing happens.

### What is Expected?

ElFocusTrap is able to detect the topest popover and always trap the focus there.

### What is actually happening?

1. Not able to select a value because focus is trapped inside the dialog
2. For nested dialogs, once the upper dialog is closed, the activeElement will be body instead of the open dialog. And then, focus trap does not work as expected.

### Additional comments

_(empty)_

**Comment by init-qy** [comment]
Maybe you forgot `v-model` in your case?

**Comment by hellodresji** [comment]
> Maybe you forgot `v-model` in your case?

Yeah, I just checked again. After I added `v-model`, it works as expected. Sorry for that.
Regarding the second point, I also checked again, maybe I misused the component again?

[Nested dialogs](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uIHRleHQgQGNsaWNrPVwib3V0ZXJWaXNpYmxlID0gdHJ1ZVwiPlxuICAgIG9wZW4gdGhlIG91dGVyIERpYWxvZ1xuICA8L2VsLWJ1dHRvbj5cblxuICA8ZWwtZGF0ZS1waWNrZXIgdi1tb2RlbD1cImRhdGVTdHJcIj48L2VsLWRhdGUtcGlja2VyPlxuICBcbiAgPGVsLWRpYWxvZyB2LW1vZGVsPVwib3V0ZXJWaXNpYmxlXCIgdGl0bGU9XCJPdXRlciBEaWFsb2dcIj5cbiAgICA8dGVtcGxhdGUgI2RlZmF1bHQ+XG4gICAgICA8ZWwtZGlhbG9nXG4gICAgICAgIHYtbW9kZWw9XCJpbm5lclZpc2libGVcIlxuICAgICAgICB3aWR0aD1cIjMwJVwiXG4gICAgICAgIHRpdGxlPVwiSW5uZXIgRGlhbG9nXCJcbiAgICAgICAgYXBwZW5kLXRvLWJvZHlcbiAgICAgIC8+XG4gICAgPC90ZW1wbGF0ZT5cbiAgICA8dGVtcGxhdGUgI2Zvb3Rlcj5cbiAgICAgIDxkaXYgY2xhc3M9XCJkaWFsb2ctZm9vdGVyXCI+XG4gICAgICAgIDxlbC1idXR0b24gQGNsaWNrPVwib3V0ZXJWaXNpYmxlID0gZmFsc2VcIj5DYW5jZWw8L2VsLWJ1dHRvbj5cbiAgICAgICAgPGVsLWJ1dHRvbiB0eXBlPVwicHJpbWFyeVwiIEBjbGljaz1cImlubmVyVmlzaWJsZSA9IHRydWVcIj5cbiAgICAgICAgICBvcGVuIHRoZSBpbm5lciBEaWFsb2dcbiAgICAgICAgPC9lbC1idXR0b24+XG4gICAgICA8L2Rpdj5cbiAgICA8L3RlbXBsYXRlPlxuICA8L2VsLWRpYWxvZz5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5cbmNvbnN0IG91dGVyVmlzaWJsZSA9IHJlZihmYWxzZSlcbmNvbnN0IGlubmVyVmlzaWJsZSA9IHJlZihmYWxzZSlcbmNvbnN0IGRhdGVTdHIgPSByZWYoXCJcIilcbjwvc2NyaXB0PlxuPHN0eWxlIHNjb3BlZD5cbi5kaWFsb2ctZm9vdGVyIGJ1dHRvbjpmaXJzdC1jaGlsZCB7XG4gIG1hcmdpbi1yaWdodDogMTBweDtcbn1cbjwvc3R5bGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

Steps to reproduce:
1. open two dialogs
![image](https://user-images.githubusercontent.com/13019006/197828793-b8c8d679-d983-4d6e-b173-41ce2a636773.png)
3. close the top one
4. press tab
![image](https://user-images.githubusercontent.com/13019006/197828873-440160a7-30e1-49dc-8c9c-afd6d80298f3.png)

**Comment by Dsaquel** [comment]
Fixed in newer versions.
